src/org/labkey/test/components/ui/grids/EditableGrid.java

```diff
@@ -351,10 +365,26 @@ public void setCellValue(int row, String columnName, Object value)
             // Treat the object being sent in as a string.
             // Get the inputCell enter the text and then make the inputCell go away (hit RETURN).
 
-            WebElement inputCell = elementCache().inputCell();
-            inputCell.clear();
+            if (useDatePicker)
```

Comment:
Consider checking `value`s type instead of adding a special flag for using the date picker
```suggestion
            if (value instanceof Date)
```

Reply:
The idea is that both keyboard input and picker selection are supported, and we don't want to switch all date field test path to use picker. But will add a format check to confirm dateStr is of yyyy-MM-dd format.